### PR TITLE
Add information requests and other review changes

### PR DESCRIPTION
**Pull request by jacek-casper**
Please consider the following when creating a PR:

* Provide a useful description of this PR, suitably verbose, aimed at helping reviewers and contributors
* Update all relevant changelogs
* Provide a link to the GitHub issue relating to this PR
* Identify if any downstream impact as in to, SDKs, SmartContracts etc

### CODE REVIEW

File: node/src/reactor/main_reactor/tests/binary_port.rs [hunk 1]
```diff
@@ -196,15 +202,20 @@ async fn binary_port_component() {
...
     ) = setup().await;
 
     let test_cases = &[
-        block_hash_2_height(),
-        highest_complete_block(),
-        completed_blocks_contain_true(),
-        completed_blocks_contain_false(),
-        transaction_hash_2_block_hash_and_height(&mut rng),
+        block_header_info(*highest_block.block().hash()),
+        signed_block_info(*highest_block.block().hash()),
```
Comment (jacek-casper) [inline node/src/reactor/main_reactor/tests/binary_port.rs:218]: These 4 were deleted:
- BlockHash2Height
- HighestCompleteBlock
- CompletedBlocksContain
- TransactionHash2BlockHashAndHeight

The reason is that during review we noticed they become redundant once we add the new `InformationRequest`, which support retrieving complete block and transaction information with the `BlockIdentifier` enum which already supports both height and hash.

File: node/src/components/binary_port/metrics.rs [hunk 2]
```diff
@@ -11,22 +11,16 @@ const BINARY_PORT_TRY_SPECULATIVE_EXEC_COUNT_NAME: &str = "binary_port_try_specu
 const BINARY_PORT_TRY_SPECULATIVE_EXEC_COUNT_HELP: &str =
     "number of TrySpeculativeExec queries received";
 
-const BINARY_PORT_GET_DB_COUNT_NAME: &str = "binary_port_get_db_count";
-const BINARY_PORT_GET_DB_COUNT_HELP: &str = "number of received Get queries related to DB";
+const BINARY_PORT_GET_RECORD_COUNT_NAME: &str = "binary_port_get_db_count";
```
Comment (rafal-ch) [inline node/src/components/binary_port/metrics.rs]:
```suggestion
const BINARY_PORT_GET_RECORD_COUNT_NAME: &str = "binary_port_get_record_count";
```

Also, please review names of other metrics.

File: node/src/components/storage/error.rs [hunk 3]
```diff
@@ -58,7 +59,7 @@ pub enum FatalStorageError {
     FailedToOverwriteBlock,
     /// Db specified in raw request has not been found in the storage module.
     #[error("unable to find db: {0}")]
-    DatabaseNotFound(DbId),
+    DatabaseNotFound(RecordId),
```
Comment (rafal-ch) [inline node/src/components/storage/error.rs:62]: `DbId` was a better fit, imho, but let it be.

File: types/src/binary_port/record_id.rs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+use serde::Serialize;
+
+#[cfg(test)]
+use crate::testing::TestRng;
+
+/// An identifier of a record type.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Serialize)]
+#[repr(u16)]
+pub enum RecordId {
+    /// Refers to `BlockHeader` db.
```
Comment (rafal-ch) [inline types/src/binary_port/record_id.rs]:
```suggestion
    /// Refers to `BlockHeader` record.
```

But not sure about this. "Record" refers to a single, well, record in a DB.

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Comment by jacek-casper** [comment]
This has been merged into the main sidecar branch. Any further reviews will happen there: https://github.com/casper-network/casper-node/pull/4491.

**Pull request merged**